bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -0,0 +1,2135 @@
+                rotelModel = RotelModel.RAP1580;
+                break;
+            case THING_TYPE_ID_RC1570:
+                rotelModel = RotelModel.RC1570;
+                break;
+            case THING_TYPE_ID_RC1572:
+                rotelModel = RotelModel.RC1572;
+                break;
+            case THING_TYPE_ID_RC1590:
+                rotelModel = RotelModel.RC1590;
+                break;
+            case THING_TYPE_ID_RCD1570:
+                rotelModel = RotelModel.RCD1570;
+                break;
+            case THING_TYPE_ID_RCD1572:
+                rotelModel = RotelModel.RCD1572;
+                break;
+            case THING_TYPE_ID_RCX1500:
+                rotelModel = RotelModel.RCX1500;
+                break;
+            case THING_TYPE_ID_RDD1580:
+                rotelModel = RotelModel.RDD1580;
+                break;
+            case THING_TYPE_ID_RDG1520:
+            case THING_TYPE_ID_RT09:
+                rotelModel = RotelModel.RDG1520;
+                break;
+            case THING_TYPE_ID_RSP1576:
+                rotelModel = RotelModel.RSP1576;
+                break;
+            case THING_TYPE_ID_RSP1582:
+                rotelModel = RotelModel.RSP1582;
+                break;
+            case THING_TYPE_ID_RT11:
+                rotelModel = RotelModel.RT11;
+                break;
+            case THING_TYPE_ID_RT1570:
+                rotelModel = RotelModel.RT1570;
+                break;
+            case THING_TYPE_ID_T11:
+                rotelModel = RotelModel.T11;
+                break;
+            case THING_TYPE_ID_T14:
+                rotelModel = RotelModel.T14;
+                break;
+            default:
+                rotelModel = DEFAULT_MODEL;
+                break;
+        }
+
+        RotelThingConfiguration config = getConfigAs(RotelThingConfiguration.class);
+
+        RotelProtocol rotelProtocol = RotelProtocol.HEX;
+        if (config.protocol != null && !config.protocol.isEmpty()) {
+            try {
+                rotelProtocol = RotelProtocol.getFromName(config.protocol);
+            } catch (RotelException e) {
+            }
+        } else {
+            Map<String, String> properties = editProperties();
+            String property = properties.get(RotelBindingConstants.PROPERTY_PROTOCOL);
+            if (property != null && !property.isEmpty()) {
+                try {
+                    rotelProtocol = RotelProtocol.getFromName(property);
+                } catch (RotelException e) {
+                }
+            }
+        }
+        logger.debug("rotelProtocol {}", rotelProtocol.getName());
+
+        Map<RotelSource, String> sourcesLabels = new HashMap<>();
+
+        connector = new RotelSimuConnector(rotelModel, rotelProtocol, sourcesLabels);
+
+        if (rotelModel.hasVolumeControl()) {
+            maxVolume = rotelModel.getVolumeMax();
+            if (!rotelModel.hasDirectVolumeControl()) {
+                logger.info(
```

Comment:
Is this log message still required, now that volume is a percentage value?

Reply:
Yes, it is for 3 models only. As you can see, there is a test around it.

Reply:
Ok, so `hasDirectVolumeControl` is true for all but the 3 exceptions then, I see.